engine/order_manager.go

```diff
@@ -552,7 +569,10 @@ func (m *OrderManager) processSubmittedOrder(newOrder *order.Submit, result orde
 // processOrders iterates over all exchange orders via API
 // and adds them to the internal order store
 func (m *OrderManager) processOrders() {
-	exchanges := m.orderStore.exchangeManager.GetExchanges()
+	exchanges, err := m.orderStore.exchangeManager.GetExchanges()
```

Comment:
Apologies for not picking up on this earlier, I think the only extra thing this feature needs is an atomic blocker here. By that I mean, if we are already running processOrders, another one won't run. The main concern I have is that if I was an aspiring trader with hundreds of active trades (😎 📈) and it takes longer than 10 seconds to process them all, we will have another `processOrders` running due to this function being called via `case <-tick.C: go m.processOrders()` and this could build up in the background as the application is running.

In order to prevent this, we can add a property to `OrderManager`: `processingOrders int32`
At the start of this function, we would have:
```
if !atomic.CompareAndSwapInt32(&m.processingOrders, 0, 1) {
		return
	}
	defer atomic.StoreInt32(&m.processingOrders, 0)
```

this then prevents any doubling up of processing orders. If you want to see what I mean, change `orderManagerDelay` to `time.Millisecond` and see how many processes are spawned